geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/list/LRemDUnitTest.java

```diff
@@ -0,0 +1,210 @@
+    String key = makeListKeyWithHashtag(1, clusterStartUp.getKeyOnServer("lrem", 1));
+    List<String> elementList = makeListWithRepeatingElements(key);
+    elementList.add(ELEMENT_TO_CHECK);
+
+    lpushPerformAndVerify(key, elementList);
+
+    Random rand = new Random();
+
+    // Check for an element that doesn't exist in the list
+    Collections.reverse(elementList);
+    assertThat(jedis.lrem(key, getCount(rand, 1), "nonExistentValue")).isEqualTo(0);
+    assertThat(jedis.lrange(key, 0, elementList.size())).isEqualTo(elementList);
+
+    // Remove all elements except for ELEMENT_TO_CHECK
+    for (int i = MAX_MATCHING_ELEMENTS - 1; i >= 0; i--) {
+      assertThat(jedis.lrem(key, getCount(rand, i), makeElementString(key, i))).isEqualTo(i);
+    }
+
+    clusterStartUp.crashVM(1); // kill primary server
+
+    assertThat(jedis.lrem(key, getCount(rand, 1), ELEMENT_TO_CHECK)).isEqualTo(1);
+    assertThat(jedis.exists(key)).isFalse();
+  }
+
+  // TODO: Need to remove ignore and modify once GEODE-10108 is done
+  @Ignore
+  @Test
+  public void givenBucketsMoveDuringLrem_thenOperationsAreNotLost() throws Exception {
+    AtomicLong runningCount = new AtomicLong(3);
+    List<String> listHashtags = makeListHashtags();
+    List<String> keys = makeListKeys(listHashtags);
+
+    List<String> elementList1 = makeListWithRepeatingElements(keys.get(0));
+    List<String> elementList2 = makeListWithRepeatingElements(keys.get(1));
+    List<String> elementList3 = makeListWithRepeatingElements(keys.get(2));
+
+    lpushPerformAndVerify(keys.get(0), elementList1);
+    lpushPerformAndVerify(keys.get(1), elementList2);
+    lpushPerformAndVerify(keys.get(2), elementList3);
+
+    int initialListSize = elementList1.size();
+    Runnable task1 =
+        () -> lremPerformAndVerify(keys.get(0), runningCount, initialListSize);
+    Runnable task2 =
+        () -> lremPerformAndVerify(keys.get(1), runningCount, initialListSize);
+    Runnable task3 =
+        () -> lremPerformAndVerify(keys.get(2), runningCount, initialListSize);
+
+    Future<Void> future1 = executor.runAsync(task1);
+    Future<Void> future2 = executor.runAsync(task2);
+    Future<Void> future3 = executor.runAsync(task3);
+
+    for (int i = 0; i < 50 && runningCount.get() > 0; i++) {
+      clusterStartUp.moveBucketForKey(listHashtags.get(i % listHashtags.size()));
+      Thread.sleep(500);
+    }
+
+    runningCount.set(0);
+
+    future1.get();
+    future2.get();
+    future3.get();
+  }
+
+  private void lremPerformAndVerify(String key, AtomicLong runningCount, int listSize) {
+    assertThat(jedis.llen(key)).isEqualTo(listSize);
+
+    Random rand = new Random();
+    long elementCount = MAX_MATCHING_ELEMENTS - 1;
+    while (jedis.llen(key) > 0 && runningCount.get() > 0) {
+      String element = makeElementString(key, (int) elementCount);
+      assertThat(jedis.lrem(key, getCount(rand, elementCount), element)).isEqualTo(elementCount);
+      elementCount--;
+    }
```

Comment:
This test could be simplified a little without losing coverage, I think. What we're really wanting to test in this case is that we don't remove too many elements, or remove the wrong elements, when buckets are moving. Making the list complex and making what we're removing from it random just makes it harder to validate rather than actually providing meaningfully different behaviour vs. just simple, repetitive operations.

It might be better to have a simpler list (a shuffled list of interleaved values of "keep" and "remove", maybe) and a fixed count that gets its sign flipped after each iteration, so we alternate removing from the head or the tail. After each execution of LREM, we should confirm both that what was returned matched what was expected (either the same as the value of count, assuming the number of matching elements is a multiple of the count, or 0 if we've removed all the matching elements) and also that the size of the list is what it should be, and the number of non-matching elements ("keep" in my example) is correct, using LRANGE. That way, we can fail as soon as we see either too many elements removed or the wrong elements removed.